Zend/zend_alloc.c

```diff
@@ -1730,10 +1730,15 @@ static void *zend_mm_alloc_huge(zend_mm_heap *heap, size_t size ZEND_FILE_LINE_D
 	void *ptr;
 
 #if ZEND_MM_LIMIT
+	if (UNEXPECTED(new_size == 0)) {
+		/* overflow in ZEND_MM_ALIGNED_SIZE_EX */
+		goto memory_limit_exhausted;
+	}
```

Comment:
On second thought, this fix doesn't look quite right ... especially the fact that this is in the `#if ZEND_MM_LIMIT` branch, while this check is necessary even without memory limit. I think it would be better to move this check outside of `#if ZEND_MM_LIMIT` and generate a "Possible integer overflow in memory allocation (%zu + %zu)" style error for that case.

Reply:
Okay, reverted for now (f2fb37a772908a9331e67f583fddcc4b1b186ccf). Will come up with a better solution ASAP.